### PR TITLE
Feature/searchbyontologylinkage

### PR DESCRIPTION
**Pull request by andrewyatz**
New feature for searching by ontology linkage. After talking to @magaliruffier it's not clear what the use-case was for this. Happy to open the PR and see where it will go.

### CODE REVIEW

**Comment by nerdstrike** [comment]
Mark might have used it in a past life. Could be useful internally

**Comment by andrewyatz** [comment]
Hmmm perhaps. I'm somewhat resistant to introducing code without a clear use case but at the same time it feels wrong to abandon possibly useful code. Grrr.

**Comment by nerdstrike** [comment]
Let's get this in. It isn't increasing our liability much, and someone in the building will probably use it.

File: modules/t/gene.t [hunk 1]
```diff
@@ -985,6 +985,19 @@ SKIP: {
...
+# Checking for External DB fetching by Ontology linkage type
+{
+  my $genes = $ga->fetch_all_by_ontology_linkage_type('GO', 'IDA');
+  is(scalar(@{$genes}), 9, 'Expect 9 genes linked to IDAs'); 
+  foreach my $gene (@{$genes}) {
+    my %linkage_types = 
+      map { $_, 1 }
+      map { @{$_->get_all_linkage_types()} }
+      @{$gene->get_all_DBLinks('GO')};
+    ok($linkage_types{'IDA'}, $gene->stable_id().' was linked to an IDA term. Seaching thorough all links we find the IDA linkage');
```
Comment (magaliruffier) [inline modules/t/gene.t]: Is that sentence even in English?
Maybe something like
'Searching through all the links until we find the IDA linkage'
or 'Searching all links thoroughly for IDA linkages'

File: modules/t/gene.t [hunk 2]
```diff
@@ -985,6 +985,19 @@ SKIP: {
...
+  }
+}
+
```
Comment (magaliruffier) [inline modules/t/gene.t]: could we have one more example with undef external_db?

File: modules/t/dbEntries.t [hunk 3]
```diff
@@ -570,6 +570,12 @@ $multi->restore();
   cmp_ok(scalar(@{$external_db_names}), '==', 111, 'Retriving all the unique External DB names');
 }
 
+# Test for fetching ids by a linkage type and database name
+{
+  my $go = $dbEntryAdaptor->get_external_db_id('GO', undef, 'no version');
+  my @gene_ids = $dbEntryAdaptor->list_gene_ids_by_external_db_id($go, 'IDA');
+  is(scalar(@gene_ids), 9, 'Expect 9 hits for genes to come back with IDAs');
+}
```
Comment (magaliruffier) [inline modules/t/dbEntries.t]: Another test with list_transcript_ids_by_external_db_id please?

**Comment by andrewyatz** [comment]
You're making me remember how to build the dev/test databases now. Unfair

**Comment by andrewyatz** [comment]
Okay changes have been pushed in.

**Comment by andrewyatz** [comment]
Failure was linked to compara not to these changes from my reading.

**Comment by nerdstrike** [comment]
Agreed. The Compara schema needs patching

**Comment by magaliruffier** [comment]
I'm having trouble with sqlite on the EBI farm, but patches should be sorted soon, then you can rebase against master and all should be good

**Comment by magaliruffier** [comment]
master is fixed now, if you rebase the test suite should pass

**Comment by magaliruffier** [comment]
@andrewyatz please rebase against master and we are good for merge

**Comment by andrewyatz** [comment]
Okay remerged

**Comment by andrewyatz** [comment]
Oh you said rebase not merge. Is that an issue?

**Comment by magaliruffier** [comment]
probably not, the branch merge should be able to clean up the history

**Comment by magaliruffier** [comment]
all good now, thanks

**Pull request merged**